### PR TITLE
#4563 Retry the connection to remote node when it fails due to tempor…

### PR DESCRIPTION
**Pull request by chia7712**
…ary issue

### Checklist
- [x] [Review Contribution Guidelines](https://ohara.readthedocs.io/en/latest/contributing.html)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#ohara-team))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #4563)
- [ ] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-agent/src/main/scala/oharastream/ohara/agent/Agent.scala [hunk 1]
```diff
@@ -116,11 +117,39 @@ object Agent {
...
+        user: String,
+        password: String,
+        retry: Int
+      ): ClientSession =
+        try client.sessionLogin(hostname, port, user, password)
+        catch {
+          case e: IllegalStateException if e.getMessage.contains("SshClient not started") && retry > 0 =>
+            Releasable.close(client)
+            client = SshClient.setUpDefaultSimpleClient()
+            trySession(hostname, port, user, password, retry - 1)
```
Comment (jackyoh) [inline ohara-agent/src/main/scala/oharastream/ohara/agent/Agent.scala:145]: 使用遞迴的方式來做 ssh retry 的動作

Comment (chia7712) [reply ohara-agent/src/main/scala/oharastream/ohara/agent/Agent.scala:145]: you are totally right :)

**Comment by jackyoh** [comment]
retry

**Comment by chia7712** [comment]
retry

**Comment by chia7712** [comment]
retry

**Comment by chia7712** [comment]
After migrating to sshd 2.4, it seems to me the connection get more stable. This PR is pending to trace the ssh connection issue.

**Comment by chia7712** [comment]
retry

**Comment by chia7712** [comment]
retry

**Comment by chia7712** [comment]
一次通過！

**Pull request merged**